s3/aide.go

```diff
@@ -94,3 +94,34 @@ func (svc *Service) SetACL(v *string) {
 func (svc *Service) SetSSE(v *string) {
 	svc.sse = v
 }
+
+// ListObjectsKeysV2Pages will list the bucket keys page-wise
+func (svc *Service) ListObjectsKeysV2Pages(params *s3.ListObjectsV2Input) ([]string, bool, error) {
+
+	var keys []string
+	var islastPage bool
+	listObjectsOutputFn := func(page *s3.ListObjectsV2Output, lastPage bool) bool {
+		if page.Contents != nil {
+			for _, obj := range page.Contents {
+				if obj.Key == nil {
+					continue
+				}
+				keys = append(keys, *obj.Key)
+			}
+			islastPage = lastPage
+		}
+		return false
+	}
+
+	err := svc.svc.ListObjectsV2Pages(params, listObjectsOutputFn)
+
+	if err != nil {
+		return nil, islastPage, err
+	}
+	return keys, islastPage, nil
+}
+
+//ListObjectsV2Input wrapper over the s3 api for getting input params
+func (svc *Service) ListObjectsV2Input() *s3.ListObjectsV2Input {
+	return &s3.ListObjectsV2Input{}
+}
```

Comment:
I'm not sure about adding a method to return an empty input. It seems superfluous.

Reply:
The intent is to allow the caller as many input params as needed to make it general for future additions. I have added some documentation and example in README.md. Hope that helps. Please let me know if any other better way you have on it.

Reply:
Documentation is always helpful. Super. Thank you for updating so quickly. I'll see if I can rustle up a merge for you.